### DISCUSSION TITLE
very nice, but Linux is a must have

### DISCUSSION THREAD

**Discussion by frankipl** [JakubMelka/PDF4QT#2] [General]
Hi
Very nice, I'm using PDF files in a few of my projects. Now it is mustache + wkhtmltopdf but I'm also signing pdf files digitally, so signature verification would be very useful. However, all my server applications runs on Linux. Can it be ported to Linux ?

Best Rergards
Marek

**Comment by JakubMelka** [comment]
Hello, I am planning to support linux next year. Now I am programming DocDiff application (diff for pdf files), then I can look on how to port it.

**Comment by JakubMelka** [comment]
Hello, Linux port should be finished (using Qbs). Please try it.

**Comment by JohnnyK007** [comment]
Dear Jakub, 
You have created a very nice project. I work with PDF files a lot. It could be primarily targeted at Linux users where similar tools are needed. Windows has its Adobe clones aplenty. Please create a .deb package, .appimage, or snap for better cross-platform Linux distribution. It should go at least under the wine. Compiling the code will discourage many potential Linux users and testers. 
Best Regards
Ján

**Reply by JakubMelka** [reply]
Hello,
I agree with you. I would like to prepare those packages. However, I am not a linux specialist and do not know how to do that. How to create deb package?

**Reply by JohnnyK007** [reply]
Dear Jakub
A .deb package would be helpful for Linux distributions based on Ubuntu or Debian. Installing from .deb will check for library dependencies. However, it can be complicated to prepare and maintain the .deb package with the arrival of new distributions and new versions of the libraries on which pdf4qt depends. A similar problem would be with the .rpm package intended for RedHat-based distributions, e.g., SuSe Linux. Until recently, Snap packages were massively promoted by Canonical with its Ubuntu, but the latest news says that Snap wants to leave Canonical. Flatpack has potential. There is a lot of talk about it, especially when it comes to Fedora distributions. Snap and Flatpack packages are independent of distributions, i.e., applications run in the "sandbox" after the package is installed. Balíky Snap and Flatpack must be installed in the system. 
Maybe, the best solution is to package the pdf4qt  to the executive package .appimage. The .appimage file can be run on any Linux distribution (i.e., old or new, Ubuntu or Fedora, etc.). There is no need to install the package .appimage. Appimage is the direct executive package. The disadvantage is the slightly larger size of the .appimage package, but not so dramatically if the .appimage package is well prepared. On the Internet, there are many instructions on how to prepare an executive .appimage package. I have yet to try to package any program. I have yet not to try to package any application. When I'm less busy, I'll try to package your project in .appimage. I'm worried about the compilation failures of pdf4qt under my current Linux due to environmental setups and library dependencies. I'm not familiar with compiling large projects. I will see. Unfortunately, your application (windows version) does not yet run in Wine. I tried it in a commercial CrossOver Linux environment.
Best Regards
Ján

**Reply by PMeira** [reply]
https://openbuildservice.org/ could be useful if native packages are required. Otherwise, I agree that either Flatpak or AppImage could work fine, besides usually resulting in larger binaries etc.

edit: just saw there's already an issue open: https://github.com/JakubMelka/PDF4QT/issues/23

**Reply by jaggeri** [reply]
Linux really needs a good PDF editor - the Snap store has nothing of high quality in this area - just lots of PDF viewers and merging/splitting

**Reply by JakubMelka** [reply]
I agree, I will try to solve the linux installation problem as soon as possible.

**Comment by JakubMelka** [comment]
Hello, @JohnnyK007 , @jaggeri , it looks someone created package for arch linux: https://aur.archlinux.org/packages/pdf4qt-git

**Comment by JakubMelka** [comment]
Debian package is now available. Closing the discussion.

**Comment by robindesvils** [comment]
Where is the debian package? 
Thanks.

**Reply by JakubMelka** [reply]
Hello, here: https://github.com/JakubMelka/PDF4QT/releases/download/v1.3.5/PDF4QT-1.3.5.deb

**Comment by raffaelemancuso** [comment]
I think this can be pinned down
